circuit/src/ecc/generic/mod.rs

```diff
@@ -1,2 +1,24 @@
-/// Native implementation
-pub mod native;
+pub struct EccAddChipset<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P>
+where
+    P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS>,
+    C::Base: FieldExt,
+    C::Scalar: FieldExt,
+{
+    p: AssignedPoint<C, N, NUM_LIMBS, NUM_BITS, P>,
+    q: AssignedPoint<C, N, NUM_LIMBS, NUM_BITS, P>,
+}
+impl<C: CurveAffine, N: FieldExt, const NUM_LIMBS: usize, const NUM_BITS: usize, P> Chipset for EccAddChipset<C, N, NUM_LIMBS, NUM_BITS, P>
+where
+    P: RnsParams<C::Base, N, NUM_LIMBS, NUM_BITS>,
+    C::Base: FieldExt,
+    C::Scalar: FieldExt,
+{
+    type Config = ...;
+    type Output = ...;
```

Comment:
These types should be specified correctly.
type Config should be `EccAddConfig`, which you should add also.
type Output should be `AssignedPoint<C, NUM_LIMBS, NUM_BITS, P>`